### PR TITLE
Direct: Adds version 3.29.0

### PR DESCRIPTION
**Pull request by imanvt**
# Pull Request Template

## Description

* Adds Direct Version 3.29.0, which upgrades System.Configuration.ConfigurationManager dependency from 4.7.0 to 6.0.0
  - This dependency version is tied to the .NET 6.0 release, which still supports .NET Standard 2.0, so it is not a breaking change.
  - This dependency was upgraded due to a security/compliance alert.
* Performance: Replaces DateTime.UtcNow.ToString("r") with Rfc1123DateTimeCache.UtcNow() 
* Fixes ObjectDisposedException during Bounded Staleness/Strong barrier requests

Micro Benchmark Test Results:

|                     Method |      Mean |     Error |    StdDev |    Median |  Gen 0 | Allocated |
|--------------------------- |----------:|----------:|----------:|----------:|-------:|----------:|
|           DateTimeToString | 58.878 ns | 1.9800 ns | 5.8070 ns | 57.634 ns | 0.0185 |      80 B |
| Rfc1123DateTimeCacheUtcNow |  2.298 ns | 0.0778 ns | 0.2280 ns |  2.228 ns |      - |         - |`

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Closing issues
Part 1 of 2 for Issue #3274

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -4,7 +4,7 @@
 		<ClientOfficialVersion>3.28.0</ClientOfficialVersion>
 		<ClientPreviewVersion>3.28.0</ClientPreviewVersion>
 		<ClientPreviewSuffixVersion>preview</ClientPreviewSuffixVersion>
-		<DirectVersion>3.28.2</DirectVersion>
+		<DirectVersion>3.29.0</DirectVersion>
```
Comment (j82w) [inline Directory.Build.props:7]: Doesn't this file needed to be updated to include the new dependencies?
https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.targets

based on this PR?
https://msdata.visualstudio.com/CosmosDB/_git/CosmosDB/pullrequest/782198

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DirectContractTests.cs [hunk 2]
```diff
@@ -107,7 +107,7 @@ public void ProjectPackageDependenciesTest()
                 { "System.Numerics.Vectors", new Version(4, 5, 0) },
                 { "Newtonsoft.Json", new Version(10, 0, 2) },
                 { "Microsoft.Bcl.AsyncInterfaces", new Version(1, 0, 0) },
-                { "System.Configuration.ConfigurationManager", new Version(4, 7, 0) },
+                { "System.Configuration.ConfigurationManager", new Version(6, 0, 0) },
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DirectContractTests.cs:110]: Please include the version change in the PR description.

File: Directory.Build.props (hunk 1 above)
Comment (ealsur) [reply Directory.Build.props:7]: We'll do it in a separate PR

**Review by FabianMeiswinkel — CHANGES_REQUESTED** [review]
Question regarding version upgrade

File: Directory.Build.props (hunk 1 above)
Comment (ealsur) [reply Directory.Build.props:7]: Reference https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3322

File: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj [hunk 3]
```diff
@@ -116,7 +116,7 @@
 		<PackageReference Include="StyleCop.Analyzers" Version="1.1.118" PrivateAssets="All" />
 
 		<!--Direct Dependencies-->
-		<PackageReference Include="System.Configuration.ConfigurationManager" Version="4.7.0" />
```
[the remark this answers is not in the document]
Comment (imanvt) [reply Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj:119]: updated the description and will make sure to add this into next set of release notes in changelog.

**Review by j82w — APPROVED** [review]

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Comment by LeszekKalibrate** [comment]
`Adds Direct Version 3.29.0, which upgrades System.Configuration.ConfigurationManager dependency from 4.7.0 to 6.0.0`
This is a breaking change when the CosmosDB SDK is used in Azure Function .Net Core 3.1.

**Pull request merged**